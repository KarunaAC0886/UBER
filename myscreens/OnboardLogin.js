import { View, Text, ScrollView, Image, TouchableOpacity } from 'react-native'
import React from 'react'
import { Ionicons } from '@expo/vector-icons';
import { FontAwesome6 } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import styles from './styles';
const OnboardLogin = () => {
    const navigation = useNavigation();
    const handleNavigate = () => {
        navigation.navigate('onboardphonenumber'); 
    };
    return (
        <ScrollView style={styles.onbo_log_main}>
            <View style={styles.onb_log_mn_cntr}>
            
            <View style={styles.onb_log_whole}>

                <Image source={require('../assets/Groupimage.png')} style={styles.onb_log_arrowimg} />
                <View style={styles.onb_acc_uber_cntr}>
                    <Text style={styles.onb_log_uber_lbl}>
                        Uber
                    </Text>
                </View>
            </View>
            <TouchableOpacity style={styles.onb_log_iconwhole}>
              
                    <Text style={styles.onb_log_muvwtsft_txt}>Move with saftey</Text>
                    <Ionicons name="shield-checkmark-outline" size={24} color="black" aria-hidden="true" style={styles.onb_log_chk_shld} />
               
            </TouchableOpacity>
            <TouchableOpacity onPress={handleNavigate} style={styles.onb_log_gtsrtd_cont}>
                
                    <Text style={styles.onb_log_gtsrtd_lbl}>Get Started</Text>
                
                <FontAwesome6  name="arrow-right-long" aria-hidden="true" style={styles.onb_log_gtsrtd_txt}></FontAwesome6>
            </TouchableOpacity>
            </View>
        </ScrollView>
    );
}
export default OnboardLogin

