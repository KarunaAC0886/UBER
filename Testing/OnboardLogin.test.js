import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import OnboardLogin from '../myscreens/OnboardLogin';

jest.mock('@expo/vector-icons', () => ({
    FontAwesome6: ' FontAwesome6',
    Ionicons:'Ionicons' 
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('onbordLogin component', () => {
  it('renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<OnboardLogin/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
