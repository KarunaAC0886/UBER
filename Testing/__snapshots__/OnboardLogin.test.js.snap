// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onbordLogin component renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#276FDD",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "height": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/Groupimage.png",
            }
          }
          style={
            {
              "height": 60,
              "width": 80,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 30,
              "borderWidth": 0,
              "elevation": 5,
              "height": undefined,
              "justifyContent": "center",
              "width": undefined,
            }
          }
        >
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 64,
                "textAlign": "center",
              }
            }
          >
            Uber
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#fff",
            "borderRadius": 16.73,
            "borderWidth": 1,
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": undefined,
            "opacity": 1,
            "padding": 10,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Move with saftey
        </Text>
        <Ionicons
          aria-hidden="true"
          color="black"
          name="shield-checkmark-outline"
          size={24}
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 26,
              "paddingLeft": undefined,
            }
          }
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#fff",
            "borderRadius": 10,
            "borderWidth": 1,
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": undefined,
            "opacity": 1,
            "paddingVertical": 5,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 25,
            }
          }
        >
          Get Started
        </Text>
        < FontAwesome6
          aria-hidden="true"
          name="arrow-right-long"
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 25,
              "paddingLeft": 15,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
